src/pages/workspace/WorkspaceInvitePage.js

```diff
@@ -65,93 +61,68 @@ const defaultProps = {
     ...policyDefaultProps,
 };
 
-class WorkspaceInvitePage extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.inviteUser = this.inviteUser.bind(this);
-        this.clearErrors = this.clearErrors.bind(this);
-        this.getExcludedUsers = this.getExcludedUsers.bind(this);
-        this.toggleOption = this.toggleOption.bind(this);
-        this.updateOptionsWithSearchTerm = this.updateOptionsWithSearchTerm.bind(this);
-        this.openPrivacyURL = this.openPrivacyURL.bind(this);
-
-        const {personalDetails, userToInvite} = OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, '', this.getExcludedUsers());
-        this.state = {
-            searchTerm: '',
-            personalDetails,
-            selectedOptions: [],
-            userToInvite,
-        };
-    }
-
-    componentDidMount() {
-        this.clearErrors();
-        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(this.props.policyMembers, this.props.personalDetails);
-        Policy.openWorkspaceInvitePage(this.props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!_.isEqual(prevProps.personalDetails, this.props.personalDetails) || !_.isEqual(prevProps.policyMembers, this.props.policyMembers)) {
-            this.updateOptionsWithSearchTerm(this.state.searchTerm);
-        }
+function WorkspaceInvitePage(props) {
+    const {translate} = useLocalize();
+    const [searchTerm, setSearchTerm] = useState('');
+    const [selectedOptions, setSelectedOptions] = useState([]);
+    const [personalDetails, setPersonalDetails] = useState([]);
+    const [userToInvite, setUserToInvite] = useState(null);
+    const openWorkspaceInvitePageRef = useRef(null);
+
+    openWorkspaceInvitePageRef.current = () => {
+        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(props.policyMembers, props.personalDetails);
+        Policy.openWorkspaceInvitePage(props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
+    };
```

Comment:
Are we fine using `useRef` for this method? AFAIK I don't see any negatives here, just a buddy check with you @pecanoro

Reply:
@pecanoro 
Please let me know your thoughts

Reply:
I'm using useEffect that way.

Reply:
Yeah, sorry, but why do you need this instead of moving it inside useEffect?

Reply:
We are using this function inside `useEffect` callback and `useOnNetworkReconnect` hook.

Reply:
Ahh I see you updated it to stop using the ref, I like it more this way!